syne_tune/optimizer/schedulers/regularized_evolution.py

```diff
@@ -0,0 +1,132 @@
+import numpy as np
+import copy
+
+from collections import deque
+from typing import Optional, Dict, List
+from dataclasses import dataclass
+
+from syne_tune.backend.trial_status import Trial
+from syne_tune.optimizer.scheduler import TrialScheduler, TrialSuggestion
+from syne_tune.search_space import Domain, Categorical, Float, Integer
+
+
+@dataclass
+class PopulationElement:
+    score: int = 0
+    config: dict = None
+
+
+class RegularizedEvolution(TrialScheduler):
+    def __init__(
+            self,
+            config_space,
+            metric: str,
+            mode: str,
+            population_size: int = 100,
+            sample_size: int = 10,
+            random_seed: int = None
+    ):
+        """
+        Implements the regularized evolution algorithm proposed by Real et al. The original implementation only
+        considers categorical hyperparameters. For integer and float parameters we sample a new value uniformly
```

Comment:
Why not rather sampling over the distribution provided in the config space? (it may be log-uniform for instance).